{"total_count": 24, "incomplete_results": false, "items": [{"url": "https://api.github.com/repos/Tinche/aiofiles/issues/80", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/80/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/80/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/80/events", "html_url": "https://github.com/Tinche/aiofiles/issues/80", "id": 654677552, "node_id": "MDU6SXNzdWU2NTQ2Nzc1NTI=", "number": 80, "title": "aiofiles.os.remove doesnt work", "user": {"login": "Lepiloff", "id": 28572064, "node_id": "MDQ6VXNlcjI4NTcyMDY0", "avatar_url": "https://avatars3.githubusercontent.com/u/28572064?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Lepiloff", "html_url": "https://github.com/Lepiloff", "followers_url": "https://api.github.com/users/Lepiloff/followers", "following_url": "https://api.github.com/users/Lepiloff/following{/other_user}", "gists_url": "https://api.github.com/users/Lepiloff/gists{/gist_id}", "starred_url": "https://api.github.com/users/Lepiloff/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Lepiloff/subscriptions", "organizations_url": "https://api.github.com/users/Lepiloff/orgs", "repos_url": "https://api.github.com/users/Lepiloff/repos", "events_url": "https://api.github.com/users/Lepiloff/events{/privacy}", "received_events_url": "https://api.github.com/users/Lepiloff/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-07-10T10:18:49Z", "updated_at": "2020-07-10T10:27:20Z", "closed_at": "2020-07-10T10:27:20Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hi! \r\nI am trying to use asynchronous file deletion with aiofiles.os.remove but the file is not deleted from the disk.\r\nCode sample\r\n```\r\nasync def create_upload_file(file: UploadFile = File(...)):\r\n    _, ext = os.path.splitext(file.filename)\r\n    content = await file.read()\r\n    if file.content_type not in ['image/jpeg', 'image/png']:\r\n        raise HTTPException(status_code=406, detail=\"Only .jpeg or .png  files allowed\")\r\n    file_name = f'{uuid.uuid4().hex}{ext}'\r\n    async with aiofiles.open(file_name, \"wb\") as f:\r\n        await f.write(content)\r\n    path_to_img = os.path.abspath(file_name)\r\n    try:\r\n        response = detect_text(path_to_img)\r\n    finally:\r\n        aiofiles.os.remove(path_to_img)\r\n    return {\"file_info\": response}\r\n```\r\n\r\nIf I change  `aiofiles.os.remove(path_to_img)` to os.remove(path_to_img) its work as expected. \r\nWhat's the reason?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/78", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/78/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/78/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/78/events", "html_url": "https://github.com/Tinche/aiofiles/issues/78", "id": 643103565, "node_id": "MDU6SXNzdWU2NDMxMDM1NjU=", "number": 78, "title": "aiofile Corrupts images", "user": {"login": "oguh43", "id": 13382592, "node_id": "MDQ6VXNlcjEzMzgyNTky", "avatar_url": "https://avatars3.githubusercontent.com/u/13382592?v=4", "gravatar_id": "", "url": "https://api.github.com/users/oguh43", "html_url": "https://github.com/oguh43", "followers_url": "https://api.github.com/users/oguh43/followers", "following_url": "https://api.github.com/users/oguh43/following{/other_user}", "gists_url": "https://api.github.com/users/oguh43/gists{/gist_id}", "starred_url": "https://api.github.com/users/oguh43/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/oguh43/subscriptions", "organizations_url": "https://api.github.com/users/oguh43/orgs", "repos_url": "https://api.github.com/users/oguh43/repos", "events_url": "https://api.github.com/users/oguh43/events{/privacy}", "received_events_url": "https://api.github.com/users/oguh43/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2020-06-22T14:09:24Z", "updated_at": "2020-06-23T22:42:16Z", "closed_at": "2020-06-23T22:07:10Z", "author_association": "NONE", "active_lock_reason": null, "body": "It seems like aiofile doesnt support writing bytes, or respect this line: `async with aiofile.AIOFile(temp,\"wb\") as f:`", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/77", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/77/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/77/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/77/events", "html_url": "https://github.com/Tinche/aiofiles/issues/77", "id": 623848033, "node_id": "MDU6SXNzdWU2MjM4NDgwMzM=", "number": 77, "title": "OSError: [Errno 24] Too many open file", "user": {"login": "FightMan01", "id": 29693718, "node_id": "MDQ6VXNlcjI5NjkzNzE4", "avatar_url": "https://avatars3.githubusercontent.com/u/29693718?v=4", "gravatar_id": "", "url": "https://api.github.com/users/FightMan01", "html_url": "https://github.com/FightMan01", "followers_url": "https://api.github.com/users/FightMan01/followers", "following_url": "https://api.github.com/users/FightMan01/following{/other_user}", "gists_url": "https://api.github.com/users/FightMan01/gists{/gist_id}", "starred_url": "https://api.github.com/users/FightMan01/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/FightMan01/subscriptions", "organizations_url": "https://api.github.com/users/FightMan01/orgs", "repos_url": "https://api.github.com/users/FightMan01/repos", "events_url": "https://api.github.com/users/FightMan01/events{/privacy}", "received_events_url": "https://api.github.com/users/FightMan01/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-05-24T11:44:03Z", "updated_at": "2020-06-04T13:08:51Z", "closed_at": "2020-06-04T13:08:51Z", "author_association": "NONE", "active_lock_reason": null, "body": "When I try to open a json in every half second I get this error.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/76", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/76/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/76/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/76/events", "html_url": "https://github.com/Tinche/aiofiles/issues/76", "id": 602686431, "node_id": "MDU6SXNzdWU2MDI2ODY0MzE=", "number": 76, "title": "awaiting for aiofiles.os.remove() returns TypeError: object NoneType can't be used in 'await' expression", "user": {"login": "OmerAbramovich", "id": 50517971, "node_id": "MDQ6VXNlcjUwNTE3OTcx", "avatar_url": "https://avatars1.githubusercontent.com/u/50517971?v=4", "gravatar_id": "", "url": "https://api.github.com/users/OmerAbramovich", "html_url": "https://github.com/OmerAbramovich", "followers_url": "https://api.github.com/users/OmerAbramovich/followers", "following_url": "https://api.github.com/users/OmerAbramovich/following{/other_user}", "gists_url": "https://api.github.com/users/OmerAbramovich/gists{/gist_id}", "starred_url": "https://api.github.com/users/OmerAbramovich/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/OmerAbramovich/subscriptions", "organizations_url": "https://api.github.com/users/OmerAbramovich/orgs", "repos_url": "https://api.github.com/users/OmerAbramovich/repos", "events_url": "https://api.github.com/users/OmerAbramovich/events{/privacy}", "received_events_url": "https://api.github.com/users/OmerAbramovich/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2020-04-19T11:04:16Z", "updated_at": "2020-04-19T11:14:38Z", "closed_at": "2020-04-19T11:14:38Z", "author_association": "NONE", "active_lock_reason": null, "body": "I'm trying to remove a file from a local directory asynchronously; however, I get the following error:\r\nobject NoneType can't be used in 'await' expression (<class 'TypeError'>)\r\nI'm using ver aiofiles 0.5.0 and Python 3.6.5\r\n\r\nmy code is as straightforward as such:\r\n\r\n```\r\nasync def delete_local_file(file_to_del):\r\n    await aiof.os.remove(file_to_del)\r\n    print(\"deleted: \"+file_to_del)\r\n\r\nawait delete_local_file(localfile)\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/74", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/74/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/74/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/74/events", "html_url": "https://github.com/Tinche/aiofiles/issues/74", "id": 599945640, "node_id": "MDU6SXNzdWU1OTk5NDU2NDA=", "number": 74, "title": "DeprecationWarning at aiofiles.os:8", "user": {"login": "HacKanCuBa", "id": 7178420, "node_id": "MDQ6VXNlcjcxNzg0MjA=", "avatar_url": "https://avatars2.githubusercontent.com/u/7178420?v=4", "gravatar_id": "", "url": "https://api.github.com/users/HacKanCuBa", "html_url": "https://github.com/HacKanCuBa", "followers_url": "https://api.github.com/users/HacKanCuBa/followers", "following_url": "https://api.github.com/users/HacKanCuBa/following{/other_user}", "gists_url": "https://api.github.com/users/HacKanCuBa/gists{/gist_id}", "starred_url": "https://api.github.com/users/HacKanCuBa/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/HacKanCuBa/subscriptions", "organizations_url": "https://api.github.com/users/HacKanCuBa/orgs", "repos_url": "https://api.github.com/users/HacKanCuBa/repos", "events_url": "https://api.github.com/users/HacKanCuBa/events{/privacy}", "received_events_url": "https://api.github.com/users/HacKanCuBa/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 8, "created_at": "2020-04-15T00:51:57Z", "updated_at": "2020-04-17T01:29:24Z", "closed_at": "2020-04-17T01:29:24Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hi there! Pytest is *correctly* complaining about the use of `@coroutine` decorator at aiofiles.os line 8. This is for the current latest 0.5.0 version.\r\n\r\nhttps://github.com/Tinche/aiofiles/blob/258e95640d6242c53a1c0d84e68343d79d80d781/aiofiles/os.py#L7-L16\r\n\r\nThe fix would be to use `async def` directly:\r\n\r\n```python\r\ndef wrap(func):\r\n    @wraps(func)\r\n    async def run(*args, loop=None, executor=None, **kwargs):\r\n        if loop is None:\r\n            loop = asyncio.get_event_loop()\r\n        pfunc = partial(func, *args, **kwargs)\r\n        return loop.run_in_executor(executor, pfunc)\r\n\r\n    return run\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/70", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/70/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/70/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/70/events", "html_url": "https://github.com/Tinche/aiofiles/issues/70", "id": 547217054, "node_id": "MDU6SXNzdWU1NDcyMTcwNTQ=", "number": 70, "title": "Python 3.8 deprecation warnings", "user": {"login": "davesteele", "id": 565205, "node_id": "MDQ6VXNlcjU2NTIwNQ==", "avatar_url": "https://avatars0.githubusercontent.com/u/565205?v=4", "gravatar_id": "", "url": "https://api.github.com/users/davesteele", "html_url": "https://github.com/davesteele", "followers_url": "https://api.github.com/users/davesteele/followers", "following_url": "https://api.github.com/users/davesteele/following{/other_user}", "gists_url": "https://api.github.com/users/davesteele/gists{/gist_id}", "starred_url": "https://api.github.com/users/davesteele/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/davesteele/subscriptions", "organizations_url": "https://api.github.com/users/davesteele/orgs", "repos_url": "https://api.github.com/users/davesteele/repos", "events_url": "https://api.github.com/users/davesteele/events{/privacy}", "received_events_url": "https://api.github.com/users/davesteele/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2020-01-09T02:52:24Z", "updated_at": "2020-06-30T23:12:32Z", "closed_at": "2020-04-11T23:23:07Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "e.g.\r\n  /home/daves/github/aiofiles/.pybuild/cpython3_3.8/build/aiofiles/threadpool/utils.py:33: DeprecationWarning: \"@coroutine\" decorator is deprecated since Python 3.8, use \"async def\" instead\r\n    def method(self, *args, **kwargs):\r\n\r\n.pybuild/cpython3_3.8/build/tests/test_simple.py::test_serve_small_bin_file_sync\r\n  /home/daves/github/aiofiles/.pybuild/cpython3_3.8/build/tests/test_simple.py:28: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.\r\n    server = yield from asyncio.start_server(serve_file, port=unused_tcp_port,\r\n\r\n\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/69", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/69/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/69/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/69/events", "html_url": "https://github.com/Tinche/aiofiles/issues/69", "id": 546949592, "node_id": "MDU6SXNzdWU1NDY5NDk1OTI=", "number": 69, "title": "Missing .name and .mode attributes", "user": {"login": "mjpieters", "id": 46775, "node_id": "MDQ6VXNlcjQ2Nzc1", "avatar_url": "https://avatars3.githubusercontent.com/u/46775?v=4", "gravatar_id": "", "url": "https://api.github.com/users/mjpieters", "html_url": "https://github.com/mjpieters", "followers_url": "https://api.github.com/users/mjpieters/followers", "following_url": "https://api.github.com/users/mjpieters/following{/other_user}", "gists_url": "https://api.github.com/users/mjpieters/gists{/gist_id}", "starred_url": "https://api.github.com/users/mjpieters/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/mjpieters/subscriptions", "organizations_url": "https://api.github.com/users/mjpieters/orgs", "repos_url": "https://api.github.com/users/mjpieters/repos", "events_url": "https://api.github.com/users/mjpieters/events{/privacy}", "received_events_url": "https://api.github.com/users/mjpieters/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2020-01-08T15:56:24Z", "updated_at": "2020-07-16T17:09:41Z", "closed_at": "2020-07-16T17:09:41Z", "author_association": "NONE", "active_lock_reason": null, "body": "The [FileIO attributes](https://docs.python.org/3/library/io.html#io.FileIO) `.name` and `.mode` are missing from the wrappers.\r\n\r\nSo this works:\r\n\r\n```python\r\nwith open(filename) as f:\r\n    print(f.name, f.mode)\r\n```\r\n\r\nThis doesn't, resulting in attribute errors.\r\n\r\n```python\r\nasync with aiofiles.open(filename) as f:\r\n    print(f.name, f.mode)\r\n```\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/67", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/67/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/67/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/67/events", "html_url": "https://github.com/Tinche/aiofiles/issues/67", "id": 532914993, "node_id": "MDU6SXNzdWU1MzI5MTQ5OTM=", "number": 67, "title": "Update pip package.", "user": {"login": "NotoriousRebel", "id": 36310667, "node_id": "MDQ6VXNlcjM2MzEwNjY3", "avatar_url": "https://avatars0.githubusercontent.com/u/36310667?v=4", "gravatar_id": "", "url": "https://api.github.com/users/NotoriousRebel", "html_url": "https://github.com/NotoriousRebel", "followers_url": "https://api.github.com/users/NotoriousRebel/followers", "following_url": "https://api.github.com/users/NotoriousRebel/following{/other_user}", "gists_url": "https://api.github.com/users/NotoriousRebel/gists{/gist_id}", "starred_url": "https://api.github.com/users/NotoriousRebel/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/NotoriousRebel/subscriptions", "organizations_url": "https://api.github.com/users/NotoriousRebel/orgs", "repos_url": "https://api.github.com/users/NotoriousRebel/repos", "events_url": "https://api.github.com/users/NotoriousRebel/events{/privacy}", "received_events_url": "https://api.github.com/users/NotoriousRebel/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2019-12-04T20:25:37Z", "updated_at": "2020-01-06T00:56:15Z", "closed_at": "2020-01-06T00:56:15Z", "author_association": "NONE", "active_lock_reason": null, "body": "I just pip installed the package and attempted to do ```aiofiles.os.mkdir```\r\nand it crashed because that function didn't exist. After debugging it appears that the pip package is missing mkdir inside os.py.  Please update the pip package. ", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/66", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/66/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/66/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/66/events", "html_url": "https://github.com/Tinche/aiofiles/issues/66", "id": 530561471, "node_id": "MDU6SXNzdWU1MzA1NjE0NzE=", "number": 66, "title": "How can i read a line and specific row", "user": {"login": "britisharmy", "id": 39626324, "node_id": "MDQ6VXNlcjM5NjI2MzI0", "avatar_url": "https://avatars1.githubusercontent.com/u/39626324?v=4", "gravatar_id": "", "url": "https://api.github.com/users/britisharmy", "html_url": "https://github.com/britisharmy", "followers_url": "https://api.github.com/users/britisharmy/followers", "following_url": "https://api.github.com/users/britisharmy/following{/other_user}", "gists_url": "https://api.github.com/users/britisharmy/gists{/gist_id}", "starred_url": "https://api.github.com/users/britisharmy/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/britisharmy/subscriptions", "organizations_url": "https://api.github.com/users/britisharmy/orgs", "repos_url": "https://api.github.com/users/britisharmy/repos", "events_url": "https://api.github.com/users/britisharmy/events{/privacy}", "received_events_url": "https://api.github.com/users/britisharmy/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-11-30T10:28:05Z", "updated_at": "2019-12-07T14:13:31Z", "closed_at": "2019-12-07T14:13:31Z", "author_association": "NONE", "active_lock_reason": null, "body": "I have this non async method\r\n\r\n```\r\n     def lastname():\r\n        with open('surnames.csv') as csvfile:\r\n             readCSV = csv.reader(csvfile, delimiter=',')\r\n             for row in readCSV:\r\n              print(row[0],row[1])\r\n```\r\n\r\nthat reads a row\r\n\r\nand i have this async\r\n\r\n```\r\n async def firstname():\r\n         async with aiofiles.open('firstnames.csv') as csvfile:\r\n              readCSV  = await csvfile.read()\r\n              return readCSV\r\n```\r\n\r\nHow can i read a row in the async method?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/65", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/65/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/65/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/65/events", "html_url": "https://github.com/Tinche/aiofiles/issues/65", "id": 448441634, "node_id": "MDU6SXNzdWU0NDg0NDE2MzQ=", "number": 65, "title": "Can we get an update release on pypi", "user": {"login": "yatmanov", "id": 23621240, "node_id": "MDQ6VXNlcjIzNjIxMjQw", "avatar_url": "https://avatars2.githubusercontent.com/u/23621240?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yatmanov", "html_url": "https://github.com/yatmanov", "followers_url": "https://api.github.com/users/yatmanov/followers", "following_url": "https://api.github.com/users/yatmanov/following{/other_user}", "gists_url": "https://api.github.com/users/yatmanov/gists{/gist_id}", "starred_url": "https://api.github.com/users/yatmanov/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yatmanov/subscriptions", "organizations_url": "https://api.github.com/users/yatmanov/orgs", "repos_url": "https://api.github.com/users/yatmanov/repos", "events_url": "https://api.github.com/users/yatmanov/events{/privacy}", "received_events_url": "https://api.github.com/users/yatmanov/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 6, "created_at": "2019-05-25T08:58:37Z", "updated_at": "2020-04-11T23:23:57Z", "closed_at": "2020-04-11T23:23:56Z", "author_association": "NONE", "active_lock_reason": null, "body": "It would be nice to get access to latest changes made in project through installing the package from pypi.", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/48", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/48/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/48/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/48/events", "html_url": "https://github.com/Tinche/aiofiles/issues/48", "id": 356298662, "node_id": "MDU6SXNzdWUzNTYyOTg2NjI=", "number": 48, "title": "Please tag 0.4.0 release in git", "user": {"login": "ignatenkobrain", "id": 2866862, "node_id": "MDQ6VXNlcjI4NjY4NjI=", "avatar_url": "https://avatars1.githubusercontent.com/u/2866862?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ignatenkobrain", "html_url": "https://github.com/ignatenkobrain", "followers_url": "https://api.github.com/users/ignatenkobrain/followers", "following_url": "https://api.github.com/users/ignatenkobrain/following{/other_user}", "gists_url": "https://api.github.com/users/ignatenkobrain/gists{/gist_id}", "starred_url": "https://api.github.com/users/ignatenkobrain/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ignatenkobrain/subscriptions", "organizations_url": "https://api.github.com/users/ignatenkobrain/orgs", "repos_url": "https://api.github.com/users/ignatenkobrain/repos", "events_url": "https://api.github.com/users/ignatenkobrain/events{/privacy}", "received_events_url": "https://api.github.com/users/ignatenkobrain/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2018-09-02T17:17:34Z", "updated_at": "2018-09-02T23:14:14Z", "closed_at": "2018-09-02T23:14:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "In Fedora we package from github tarball rather than PyPI, so this would help us updating to 0.4.0.\r\n\r\nThanks!", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/42", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/42/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/42/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/42/events", "html_url": "https://github.com/Tinche/aiofiles/issues/42", "id": 336510183, "node_id": "MDU6SXNzdWUzMzY1MTAxODM=", "number": 42, "title": "Python 3.7", "user": {"login": "Modelmat", "id": 25718989, "node_id": "MDQ6VXNlcjI1NzE4OTg5", "avatar_url": "https://avatars1.githubusercontent.com/u/25718989?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Modelmat", "html_url": "https://github.com/Modelmat", "followers_url": "https://api.github.com/users/Modelmat/followers", "following_url": "https://api.github.com/users/Modelmat/following{/other_user}", "gists_url": "https://api.github.com/users/Modelmat/gists{/gist_id}", "starred_url": "https://api.github.com/users/Modelmat/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Modelmat/subscriptions", "organizations_url": "https://api.github.com/users/Modelmat/orgs", "repos_url": "https://api.github.com/users/Modelmat/repos", "events_url": "https://api.github.com/users/Modelmat/events{/privacy}", "received_events_url": "https://api.github.com/users/Modelmat/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2018-06-28T07:47:19Z", "updated_at": "2018-08-11T17:09:51Z", "closed_at": "2018-08-11T17:09:51Z", "author_association": "NONE", "active_lock_reason": null, "body": "Are there any known issues running in python 3.7?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/39", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/39/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/39/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/39/events", "html_url": "https://github.com/Tinche/aiofiles/issues/39", "id": 317138875, "node_id": "MDU6SXNzdWUzMTcxMzg4NzU=", "number": 39, "title": "Many writes to a file", "user": {"login": "sander76", "id": 1688712, "node_id": "MDQ6VXNlcjE2ODg3MTI=", "avatar_url": "https://avatars2.githubusercontent.com/u/1688712?v=4", "gravatar_id": "", "url": "https://api.github.com/users/sander76", "html_url": "https://github.com/sander76", "followers_url": "https://api.github.com/users/sander76/followers", "following_url": "https://api.github.com/users/sander76/following{/other_user}", "gists_url": "https://api.github.com/users/sander76/gists{/gist_id}", "starred_url": "https://api.github.com/users/sander76/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/sander76/subscriptions", "organizations_url": "https://api.github.com/users/sander76/orgs", "repos_url": "https://api.github.com/users/sander76/repos", "events_url": "https://api.github.com/users/sander76/events{/privacy}", "received_events_url": "https://api.github.com/users/sander76/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2018-04-24T09:16:50Z", "updated_at": "2018-04-24T09:26:24Z", "closed_at": "2018-04-24T09:26:23Z", "author_association": "NONE", "active_lock_reason": null, "body": "I want to write to a file many times during execution of my code.\r\n\r\nLooking at the examples and the code a thread is created using `run_in_executor` during the `open` method of `aiofiles`.\r\nSo each time I want to write to the file a new thread is created. Which is expensive I assume ? (am I now developing for a raspberrypi)\r\n\r\nIs it possible to dedicate a specific thread to a specific file ? Or should I keep the file open all the time. Do my writes and finally close it ?\r\n\r\nThanks !", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/38", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/38/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/38/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/38/events", "html_url": "https://github.com/Tinche/aiofiles/issues/38", "id": 304571679, "node_id": "MDU6SXNzdWUzMDQ1NzE2Nzk=", "number": 38, "title": "`async with` throws error", "user": {"login": "ek-nath", "id": 1240693, "node_id": "MDQ6VXNlcjEyNDA2OTM=", "avatar_url": "https://avatars0.githubusercontent.com/u/1240693?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ek-nath", "html_url": "https://github.com/ek-nath", "followers_url": "https://api.github.com/users/ek-nath/followers", "following_url": "https://api.github.com/users/ek-nath/following{/other_user}", "gists_url": "https://api.github.com/users/ek-nath/gists{/gist_id}", "starred_url": "https://api.github.com/users/ek-nath/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ek-nath/subscriptions", "organizations_url": "https://api.github.com/users/ek-nath/orgs", "repos_url": "https://api.github.com/users/ek-nath/repos", "events_url": "https://api.github.com/users/ek-nath/events{/privacy}", "received_events_url": "https://api.github.com/users/ek-nath/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2018-03-12T23:03:54Z", "updated_at": "2018-03-14T20:42:32Z", "closed_at": "2018-03-14T20:42:32Z", "author_association": "NONE", "active_lock_reason": null, "body": "<pre><code>import aiofiles\r\nprint(aiofiles.__version__)\r\n\r\nasync with aiofiles.open('wctDA2.tmp', mode='r') as f:\r\n    contents = await f.read()\r\nprint(contents)\r\n</code></pre>\r\n\r\nOutput:\r\n<pre><code>\r\nFile \"a.py\", line 4\r\n    async with aiofiles.open('wctDA2.tmp', mode='r') as f:\r\n             ^\r\nSyntaxError: invalid syntax\r\n</code></pre>\r\n\r\nDetails:\r\n\r\n1. python: Python 3.6.4 :: Anaconda custom (64-bit)\r\n2. OS: Windows 10 x64\r\n3. aiofiles: 0.3.2", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/30", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/30/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/30/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/30/events", "html_url": "https://github.com/Tinche/aiofiles/issues/30", "id": 259990743, "node_id": "MDU6SXNzdWUyNTk5OTA3NDM=", "number": 30, "title": "import error", "user": {"login": "Rouzip", "id": 24493603, "node_id": "MDQ6VXNlcjI0NDkzNjAz", "avatar_url": "https://avatars1.githubusercontent.com/u/24493603?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Rouzip", "html_url": "https://github.com/Rouzip", "followers_url": "https://api.github.com/users/Rouzip/followers", "following_url": "https://api.github.com/users/Rouzip/following{/other_user}", "gists_url": "https://api.github.com/users/Rouzip/gists{/gist_id}", "starred_url": "https://api.github.com/users/Rouzip/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Rouzip/subscriptions", "organizations_url": "https://api.github.com/users/Rouzip/orgs", "repos_url": "https://api.github.com/users/Rouzip/repos", "events_url": "https://api.github.com/users/Rouzip/events{/privacy}", "received_events_url": "https://api.github.com/users/Rouzip/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2017-09-23T07:42:55Z", "updated_at": "2017-09-23T09:26:48Z", "closed_at": "2017-09-23T09:26:48Z", "author_association": "NONE", "active_lock_reason": null, "body": "when I import aiofile,it will raise a OS Error like that:\r\n```python\r\nPython 3.5.2 (default, Nov 17 2016, 17:05:23) \r\n[GCC 5.4.0 20160609] on linux\r\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\r\n>>> import aiofile\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"/home/rouzip/.local/lib/python3.5/site-packages/aiofile/__init__.py\", line 1, in <module>\r\n    from .utils import Reader, Writer\r\n  File \"/home/rouzip/.local/lib/python3.5/site-packages/aiofile/utils.py\", line 3, in <module>\r\n    from .aio import AIOFile\r\n  File \"/home/rouzip/.local/lib/python3.5/site-packages/aiofile/aio.py\", line 6, in <module>\r\n    from .posix_aio import IO_NOP, IO_WRITE, IO_READ, AIOOperation\r\n  File \"aiofile/posix_aio.pyx\", line 67, in init aiofile.posix_aio (aiofile/posix_aio.c:7366)\r\nOSError: [Errno 22] Invalid argument\r\n```\r\nI don't know what should I do to fix this problem,can anybody help me?\r\nI use python3.5.2 and my os is ubuntu16.04", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/26", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/26/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/26/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/26/events", "html_url": "https://github.com/Tinche/aiofiles/issues/26", "id": 241216921, "node_id": "MDU6SXNzdWUyNDEyMTY5MjE=", "number": 26, "title": "Doesn't save file", "user": {"login": "kobzar", "id": 1912842, "node_id": "MDQ6VXNlcjE5MTI4NDI=", "avatar_url": "https://avatars2.githubusercontent.com/u/1912842?v=4", "gravatar_id": "", "url": "https://api.github.com/users/kobzar", "html_url": "https://github.com/kobzar", "followers_url": "https://api.github.com/users/kobzar/followers", "following_url": "https://api.github.com/users/kobzar/following{/other_user}", "gists_url": "https://api.github.com/users/kobzar/gists{/gist_id}", "starred_url": "https://api.github.com/users/kobzar/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/kobzar/subscriptions", "organizations_url": "https://api.github.com/users/kobzar/orgs", "repos_url": "https://api.github.com/users/kobzar/repos", "events_url": "https://api.github.com/users/kobzar/events{/privacy}", "received_events_url": "https://api.github.com/users/kobzar/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2017-07-07T10:00:40Z", "updated_at": "2017-07-26T10:24:58Z", "closed_at": "2017-07-26T10:24:58Z", "author_association": "NONE", "active_lock_reason": null, "body": "I have standard method to save file. \r\nBut when i use \r\n```\r\n        with open(file_path, 'wb') as fsave:\r\n            fsave.write(file_data)\r\n\r\n```\r\nFiles saved without errors. \r\nIf i use \r\n```\r\n        async with aiofiles.open(file_path, 'wb') as fsave:\r\n            fsave.write(file_data)\r\n\r\n```\r\nFiles created, but with 0 size.  file_data the same.\r\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/23", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/23/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/23/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/23/events", "html_url": "https://github.com/Tinche/aiofiles/issues/23", "id": 234413214, "node_id": "MDU6SXNzdWUyMzQ0MTMyMTQ=", "number": 23, "title": "Test fails for sendfile with EINVAL", "user": {"login": "nehaljwani", "id": 1779189, "node_id": "MDQ6VXNlcjE3NzkxODk=", "avatar_url": "https://avatars0.githubusercontent.com/u/1779189?v=4", "gravatar_id": "", "url": "https://api.github.com/users/nehaljwani", "html_url": "https://github.com/nehaljwani", "followers_url": "https://api.github.com/users/nehaljwani/followers", "following_url": "https://api.github.com/users/nehaljwani/following{/other_user}", "gists_url": "https://api.github.com/users/nehaljwani/gists{/gist_id}", "starred_url": "https://api.github.com/users/nehaljwani/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/nehaljwani/subscriptions", "organizations_url": "https://api.github.com/users/nehaljwani/orgs", "repos_url": "https://api.github.com/users/nehaljwani/repos", "events_url": "https://api.github.com/users/nehaljwani/events{/privacy}", "received_events_url": "https://api.github.com/users/nehaljwani/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2017-06-08T05:11:08Z", "updated_at": "2017-09-08T20:06:00Z", "closed_at": "2017-09-08T20:06:00Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "```python\r\n==================================================================================================== FAILURES =====================================================================================================\r\n_______________________________________________________________________________________________ test_sendfile_file ________________________________________________________________________________________________\r\n\r\ntmpdir = local('/tmp/pytest-of-nwani/pytest-6/test_sendfile_file0')\r\n\r\n    @pytest.mark.asyncio\r\n    def test_sendfile_file(tmpdir):\r\n        \"\"\"Test the sendfile functionality, file-to-file.\"\"\"\r\n        filename = join(dirname(__file__), 'resources', 'test_file1.txt')\r\n        tmp_filename = tmpdir.join('tmp.bin')\r\n\r\n        with open(filename) as f:\r\n            contents = f.read()\r\n\r\n        input_file = yield from aiofiles.open(filename)\r\n        output_file = yield from aiofiles.open(str(tmp_filename), mode='w+')\r\n\r\n        size = (yield from aiofiles.os.stat(filename)).st_size\r\n\r\n        input_fd = input_file.fileno()\r\n        output_fd = output_file.fileno()\r\n\r\n>       yield from aiofiles.os.sendfile(output_fd, input_fd, 0, size)\r\n\r\ntests/test_os.py:35:\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\n../_t_env/lib/python3.6/asyncio/coroutines.py:213: in coro\r\n    res = yield from res\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\n\r\nself = <concurrent.futures.thread._WorkItem object at 0x2af170783ef0>\r\n\r\n    def run(self):\r\n        if not self.future.set_running_or_notify_cancel():\r\n            return\r\n\r\n        try:\r\n>           result = self.fn(*self.args, **self.kwargs)\r\nE           OSError: [Errno 22] Invalid argument\r\n\r\n../_t_env/lib/python3.6/concurrent/futures/thread.py:55: OSError\r\n====================================================================================== 1 failed, 229 passed in 6.33 seconds =======================================================================================\r\n```\r\n\r\nStrace to the rescue:\r\n\r\n```bash\r\n669   sendfile(19, 18, [0], 10)         = -1 EINVAL (Invalid argument)\r\n```", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/21", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/21/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/21/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/21/events", "html_url": "https://github.com/Tinche/aiofiles/issues/21", "id": 229439715, "node_id": "MDU6SXNzdWUyMjk0Mzk3MTU=", "number": 21, "title": "pickle support", "user": {"login": "ayharano", "id": 13336, "node_id": "MDQ6VXNlcjEzMzM2", "avatar_url": "https://avatars1.githubusercontent.com/u/13336?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ayharano", "html_url": "https://github.com/ayharano", "followers_url": "https://api.github.com/users/ayharano/followers", "following_url": "https://api.github.com/users/ayharano/following{/other_user}", "gists_url": "https://api.github.com/users/ayharano/gists{/gist_id}", "starred_url": "https://api.github.com/users/ayharano/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ayharano/subscriptions", "organizations_url": "https://api.github.com/users/ayharano/orgs", "repos_url": "https://api.github.com/users/ayharano/repos", "events_url": "https://api.github.com/users/ayharano/events{/privacy}", "received_events_url": "https://api.github.com/users/ayharano/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 6, "created_at": "2017-05-17T17:59:36Z", "updated_at": "2018-07-18T22:51:00Z", "closed_at": "2018-07-18T22:51:00Z", "author_association": "NONE", "active_lock_reason": null, "body": "Similar to issue #20 , how can standard [pickle](https://docs.python.org/3/library/pickle.html) interface be used along with aiofiles? How could it be done? By mimicking pickle sourcecode equivalent into aiofiles? Can such support be a feature of aiofiles?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/16", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/16/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/16/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/16/events", "html_url": "https://github.com/Tinche/aiofiles/issues/16", "id": 211002418, "node_id": "MDU6SXNzdWUyMTEwMDI0MTg=", "number": 16, "title": "creating tests with fake files", "user": {"login": "buhman", "id": 728108, "node_id": "MDQ6VXNlcjcyODEwOA==", "avatar_url": "https://avatars0.githubusercontent.com/u/728108?v=4", "gravatar_id": "", "url": "https://api.github.com/users/buhman", "html_url": "https://github.com/buhman", "followers_url": "https://api.github.com/users/buhman/followers", "following_url": "https://api.github.com/users/buhman/following{/other_user}", "gists_url": "https://api.github.com/users/buhman/gists{/gist_id}", "starred_url": "https://api.github.com/users/buhman/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/buhman/subscriptions", "organizations_url": "https://api.github.com/users/buhman/orgs", "repos_url": "https://api.github.com/users/buhman/repos", "events_url": "https://api.github.com/users/buhman/events{/privacy}", "received_events_url": "https://api.github.com/users/buhman/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2017-03-01T07:39:42Z", "updated_at": "2017-03-13T05:49:24Z", "closed_at": "2017-03-13T05:48:54Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "I noticed aiofiles' unit tests use real file IO.\r\n\r\nIn my project that uses aiofiles, I think it might make more sense to unit test with faked file IO, using something like [pyfakefs](https://github.com/jmcgeheeiv/pyfakefs).\r\n\r\nThe setup code I wrote for this feels a little clunky, but it seems to work:\r\n\r\n```python\r\n@pytest.fixture\r\ndef fs(request):\r\n    \"\"\" Fake filesystem. \"\"\"\r\n    patcher = Patcher()\r\n    patcher.setUp()\r\n\r\n    patcher._stubs.SmartSet(threadpool, '_sync_open', patcher.fake_open)\r\n\r\n    request.addfinalizer(patcher.tearDown)\r\n    return patcher.fs\r\n\r\n\r\n@threadpool.wrap.register(FakeFileWrapper)\r\ndef _(file, *, loop=None, executor=None):\r\n    return AsyncBufferedIOBase(file, loop=loop, executor=executor)\r\n```\r\n\r\nSo there are two problems that this code solves:\r\n\r\n - `Patcher` monkey-patches `builtins.open`, but because aiofiles already made a reference to `open`, this doesn't get replaced (contrast to referencing `builtins.open` instead of `_sync_open`)\r\n - `FakeFileWrapper` from pyfakefs I think probably should be a subclass or registered class of `io.FileIO` or similar.\r\n\r\nI am unsure if I should either:\r\n\r\n - directly \"fix\" these problems in aiofiles and pyfakefs respectively, or\r\n - make a small pytest plugin that handles this like the code example above.\r\n\r\nWhat do you think?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/15", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/15/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/15/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/15/events", "html_url": "https://github.com/Tinche/aiofiles/issues/15", "id": 209835985, "node_id": "MDU6SXNzdWUyMDk4MzU5ODU=", "number": 15, "title": "nonblocking IO", "user": {"login": "buhman", "id": 728108, "node_id": "MDQ6VXNlcjcyODEwOA==", "avatar_url": "https://avatars0.githubusercontent.com/u/728108?v=4", "gravatar_id": "", "url": "https://api.github.com/users/buhman", "html_url": "https://github.com/buhman", "followers_url": "https://api.github.com/users/buhman/followers", "following_url": "https://api.github.com/users/buhman/following{/other_user}", "gists_url": "https://api.github.com/users/buhman/gists{/gist_id}", "starred_url": "https://api.github.com/users/buhman/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/buhman/subscriptions", "organizations_url": "https://api.github.com/users/buhman/orgs", "repos_url": "https://api.github.com/users/buhman/repos", "events_url": "https://api.github.com/users/buhman/events{/privacy}", "received_events_url": "https://api.github.com/users/buhman/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2017-02-23T17:53:26Z", "updated_at": "2017-02-23T21:48:02Z", "closed_at": "2017-02-23T18:15:02Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "I've been comparing how aiofiles does things compared to other libraries like aiohttp.\r\n\r\naiofiles runs blocking functions in threadpools using `loop.run_in_executor`, while aiohttp mostly sets file descriptors to nonblocking mode, then uses `loop.add_writer`/`loop.add_reader` as necessary.\r\n\r\nWhy did aiofiles choose the former approach?", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/12", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/12/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/12/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/12/events", "html_url": "https://github.com/Tinche/aiofiles/issues/12", "id": 198268121, "node_id": "MDU6SXNzdWUxOTgyNjgxMjE=", "number": 12, "title": "inconsistency between version & tag", "user": {"login": "ignatenkobrain", "id": 2866862, "node_id": "MDQ6VXNlcjI4NjY4NjI=", "avatar_url": "https://avatars1.githubusercontent.com/u/2866862?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ignatenkobrain", "html_url": "https://github.com/ignatenkobrain", "followers_url": "https://api.github.com/users/ignatenkobrain/followers", "following_url": "https://api.github.com/users/ignatenkobrain/following{/other_user}", "gists_url": "https://api.github.com/users/ignatenkobrain/gists{/gist_id}", "starred_url": "https://api.github.com/users/ignatenkobrain/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ignatenkobrain/subscriptions", "organizations_url": "https://api.github.com/users/ignatenkobrain/orgs", "repos_url": "https://api.github.com/users/ignatenkobrain/repos", "events_url": "https://api.github.com/users/ignatenkobrain/events{/privacy}", "received_events_url": "https://api.github.com/users/ignatenkobrain/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2017-01-01T13:59:12Z", "updated_at": "2017-02-14T23:58:30Z", "closed_at": "2017-02-14T23:58:30Z", "author_association": "NONE", "active_lock_reason": null, "body": "Version is 0.3.0, but tag is 0.3... Would be nice to give some consistency ;)", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/6", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/6/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/6/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/6/events", "html_url": "https://github.com/Tinche/aiofiles/issues/6", "id": 174808917, "node_id": "MDU6SXNzdWUxNzQ4MDg5MTc=", "number": 6, "title": "Package cannot be installed via `pip --no-binaries`", "user": {"login": "podshumok", "id": 314085, "node_id": "MDQ6VXNlcjMxNDA4NQ==", "avatar_url": "https://avatars0.githubusercontent.com/u/314085?v=4", "gravatar_id": "", "url": "https://api.github.com/users/podshumok", "html_url": "https://github.com/podshumok", "followers_url": "https://api.github.com/users/podshumok/followers", "following_url": "https://api.github.com/users/podshumok/following{/other_user}", "gists_url": "https://api.github.com/users/podshumok/gists{/gist_id}", "starred_url": "https://api.github.com/users/podshumok/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/podshumok/subscriptions", "organizations_url": "https://api.github.com/users/podshumok/orgs", "repos_url": "https://api.github.com/users/podshumok/repos", "events_url": "https://api.github.com/users/podshumok/events{/privacy}", "received_events_url": "https://api.github.com/users/podshumok/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2016-09-02T17:22:25Z", "updated_at": "2016-09-03T19:43:56Z", "closed_at": "2016-09-03T19:43:56Z", "author_association": "NONE", "active_lock_reason": null, "body": "I prefer to not install binary packages like wheels for libraries, that do not require compilation during installation.\nBut there is no .gz package in PIPI, so `pip install --no-binaries :all: aiofiles` will fail.\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/5", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/5/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/5/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/5/events", "html_url": "https://github.com/Tinche/aiofiles/issues/5", "id": 169807034, "node_id": "MDU6SXNzdWUxNjk4MDcwMzQ=", "number": 5, "title": "Atomic writes with aiofiles", "user": {"login": "GMLudo", "id": 243823, "node_id": "MDQ6VXNlcjI0MzgyMw==", "avatar_url": "https://avatars1.githubusercontent.com/u/243823?v=4", "gravatar_id": "", "url": "https://api.github.com/users/GMLudo", "html_url": "https://github.com/GMLudo", "followers_url": "https://api.github.com/users/GMLudo/followers", "following_url": "https://api.github.com/users/GMLudo/following{/other_user}", "gists_url": "https://api.github.com/users/GMLudo/gists{/gist_id}", "starred_url": "https://api.github.com/users/GMLudo/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/GMLudo/subscriptions", "organizations_url": "https://api.github.com/users/GMLudo/orgs", "repos_url": "https://api.github.com/users/GMLudo/repos", "events_url": "https://api.github.com/users/GMLudo/events{/privacy}", "received_events_url": "https://api.github.com/users/GMLudo/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2016-08-07T17:37:32Z", "updated_at": "2016-08-07T19:08:25Z", "closed_at": "2016-08-07T19:08:25Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "Hi,\n\nFor a project, I need to have atomic writes for files.\nI've found this library: https://github.com/untitaker/python-atomicwrites\n\nI don't see an obvious way to integrate python-atomicwrites with aiofiles.\n\nWhat's your recommended approach ?\nUse python-atomicwrites with a threadpool, or fork aiofiles to use python-atomicwrites ?\nOther approach ?\n\nHave a nice day.\n", "performed_via_github_app": null, "score": 1.0}, {"url": "https://api.github.com/repos/Tinche/aiofiles/issues/3", "repository_url": "https://api.github.com/repos/Tinche/aiofiles", "labels_url": "https://api.github.com/repos/Tinche/aiofiles/issues/3/labels{/name}", "comments_url": "https://api.github.com/repos/Tinche/aiofiles/issues/3/comments", "events_url": "https://api.github.com/repos/Tinche/aiofiles/issues/3/events", "html_url": "https://github.com/Tinche/aiofiles/issues/3", "id": 135406564, "node_id": "MDU6SXNzdWUxMzU0MDY1NjQ=", "number": 3, "title": "Asynchonous context managers and iterators", "user": {"login": "pstch", "id": 1682742, "node_id": "MDQ6VXNlcjE2ODI3NDI=", "avatar_url": "https://avatars1.githubusercontent.com/u/1682742?v=4", "gravatar_id": "", "url": "https://api.github.com/users/pstch", "html_url": "https://github.com/pstch", "followers_url": "https://api.github.com/users/pstch/followers", "following_url": "https://api.github.com/users/pstch/following{/other_user}", "gists_url": "https://api.github.com/users/pstch/gists{/gist_id}", "starred_url": "https://api.github.com/users/pstch/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/pstch/subscriptions", "organizations_url": "https://api.github.com/users/pstch/orgs", "repos_url": "https://api.github.com/users/pstch/repos", "events_url": "https://api.github.com/users/pstch/events{/privacy}", "received_events_url": "https://api.github.com/users/pstch/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 7, "created_at": "2016-02-22T12:43:52Z", "updated_at": "2016-02-24T23:50:35Z", "closed_at": "2016-02-24T23:50:35Z", "author_association": "NONE", "active_lock_reason": null, "body": "`README.md` says :\n\n> The closing of a file may block, and yielding from a coroutine while exiting from a context manager isn't possible, so aiofiles file objects can't be used as context managers. Use the try/finally construct from the introductory section to ensure files are closed.\n> \n> Iteration is also unsupported. To iterate over a file, call readline repeatedly until an empty result is returned. Keep in mind readline doesn't strip newline characters.\n\nPython 3.5 provides asynchronous context managers and iterators, using `__aenter__`, `__aexit__`, `__aiter__` and `__anext__`. I think adding support for these could provide a nice API for aiofiles, and it wouldn'bt be harder than something along the lines of:\n\n```\nclass IterableContextManagerFileWrapper:  # yes it needs a new name\n    def __init__(self, *args, **kwargs):\n        # store arguments so that file can be created in __aenter__\n        self.__args, self.__kwargs = args, kwargs\n        self.__file = None  # set in __aenter__\n\n     def __getattr__(self, name):\n         assert self.__file is not None, \"file context manager not entered\"\n         return getattr(self.__file, name)   # wrap the async file object\n\n     async def __aenter__(self):\n         self.__file = await open(*self.args, **self.kwargs)\n         return self  # return self and not file so that the value can be used as iterable\n\n     async def __aexit__(self, exc_type, exc_value, exc_tb):\n         await self.close()\n         return False  # no reason to intercept exception\n\n     async def __aiter__(self):\n         return self  # this object is an iterator, so just has to return itself to be an iterable\n\n     async def __anext__(self):\n          line = await self.readline()\n          if not line:  # EOF\n              raise StopAsyncIteration  # not StopIteration !\n          else:\n              return line\n```\n\nThe resulting wrapper could now be used as :\n\n```\nasync with IterableContextManagerFileWrapper(\"/tmp/lol\") as aiofile:\n    async for line in aiofile:\n        line = line.rstrip()\n        ...\n```\n\nMy proposed POC has a weird name because I don't really know to integrate it into the class hierarchy, so I made a wrapper that provides the proposed features on top of the objects returned by `aiofiles` coroutines.\n\nThis is also why I submit this is an issue and not a PR, because I would like to get your opinion on what would be the best way to implement.\n", "performed_via_github_app": null, "score": 1.0}]}